libs/executors/include/hpx/executors/guided_pool_executor.hpp

```diff
@@ -231,29 +224,30 @@ namespace hpx { namespace parallel { namespace execution {
         friend struct guided_pool_executor_shim;
 
     public:
-        guided_pool_executor(const std::string& pool_name, bool hp_sync = false)
-          : pool_(&resource::get_thread_pool(pool_name))
+        guided_pool_executor(
+            threads::thread_pool_base* pool, bool hp_sync = false)
+          : pool_(pool)
```

Comment:
@biddisco are you ok with this change? The constructor takes an actual thread pool instead of the name of the thread pool, which allows this not to depend on the runtime (threadmanager, partitioner) to ask for a pool with a specific name. The equivalent functionality after this change is `guided_pool_executor(&hpx::resource::get_thread_pool("pool-name"))`.

I would also deprecate `pool_executor` in favor of `thread_pool_executor(get_thread_pool("pool-name"))`. I will for the next release move `pool_executor` to the runtime module (as it's a wrapper enabled by having the runtime), and add it to the `hpx/include/*executors.hpp` headers.

Do you think the `guided_pool_executor` needs a compatibility option before I make this change (I know the linear algebra people use `pool_executor` which is why I'd like to have the compatibility option.)

Reply:
I will also move, but not deprecate, the service executors (`main_pool_executor` etc.) to the runtime module since they cannot exist without the runtime.

Reply:
I'm fine with changing the type from a name to a pool, this is not a critical path and the changes are easy. I don't thinl a compatibility layer is worth the effort as this code is barely used outside of our group.